rfcs/modal-component.md

```diff
@@ -0,0 +1,154 @@
+# Modal Component
+
+### Exports
+
+* `Modal`
+* `ModalHeader`
+* `ModalBody`
+* `ModalFooter`
+* `StyledRoot`
+* `StyledBackdrop`
+* `StyledDialog`
+* `StyledDialogInner`
+* `StyledHeader`
+* `StyledBody`
+* `StyledFooter`
+* `StyledFooterAction`
+* `StyledClose`
+* `SIZE`
+* `ROLE`
+
+### <Modal/> API
+
+* `children: React.node | ({close: fn}) => React.node` - Required
+  Modal content. The children-as-function API may be preferable for performance reasons (wont render until opened) and also gives users access to a `close()` callback that they can add to buttons
+* `isOpen: boolean` - Required
+  Controls whether the modal is visible or not
+* `closable: boolean` - Optional, Defaults to `true`
+  Whether the modal should be closable by the user (either via escape, backdrop click, etc). You can set this to `false` if your modal has an action that the user _must_ take before closing.
+* `onClose: function` - Optional
+  A callback that is invoked when the modal will close
+* `size: SIZE.{small|medium|large|full|cover|auto}|string|number`, Defaults to SIZE.default
+  Controls the size of the modal (primarily width). Can be a SIZE constant or css `width` property value.
+* `aria-labelledby: string` - Optional, Default to ''
+  If specified, will be passed to the dialog element
+* `aria-describedby: string` - Optional, Defaults to ''
+  If specified, will be passed to the dialog element
+* `aria-label: string` - Optional, Defaults to ''
+  If specified, will be passed to the dialog element (described the modal for screen readers)
+* `id: string` - Optional, Defaults to ''
+  If specified, will be passed to the dialog element
+* `role: 'dialog'|'alertdialog'|string` - Optional, Defaults to 'dialog'
+  Which accessibility role this modal should have
+* `mountNode: HTMLElement` - Optional, Defaults to document.body
+  Where to mount the modal
+* `overrides: {Root, Backdrop, Dialog, DialogInner, Close}` - Optional
+  Overrides for presentational components. See "Presentational Components Props API" below.
+  * `[ComponentName]: ReactComponent | {props: {}, style: {}, component: ReactComponent}` - Optional
+
+### <ModalHeader/> API
+
+TBD
+
+### <ModalBody/> API
+
+TBD
+
+### <ModalFooter/> API
+
+TBD
+
+### SIZE Constant
+
+* `small` - 400px width
+* `medium` - 550px width
+* `large` - 800px width
+* `full` - Span full viewport, minus some margin
+* `cover` - Span full viewport, no margin
```

Comment:
we may combine with `full` as it's just a matter of one style

Reply:
so people will just use overrides and set `margin: 0`? that could work.

Reply:
yes